src/app/organisation/editorganisation/post.controller.js

```diff
@@ -19,13 +19,15 @@ module.exports = (req, res) => {
   validator.validateChains([orgNameChain])
     .then(() => {
       // API should return OrgId
-      cookie.setOrganisationName(req.body.orgName);
       organisationApi.update(orgName, cookie.getOrganisationId())
         .then((apiResponse) => {
           const responseObj = JSON.parse(apiResponse);
           logger.debug(`Response from API: ${JSON.stringify(responseObj)}`);
           // Check for error
-          res.redirect('/organisation');
+          cookie.setOrganisationName(req.body.orgName);
+          req.session.save(() => {
```

Comment:
Sometimes, the E2E tests would perform edit organisation save operations quickly enough that the session is not updated before the test assertions.

Added the req.session.save with the redirect as the callback to ensure the page returns only after the session is saved (and the cookie value is set).